base/loggers/OwnPatternFormatter.cpp

```diff
@@ -7,13 +7,118 @@
 #include <Interpreters/InternalTextLogsQueue.h>
 #include <Common/CurrentThread.h>
 #include <base/terminalColors.h>
-
+#include "Loggers.h"
 
 OwnPatternFormatter::OwnPatternFormatter(bool color_)
     : Poco::PatternFormatter(""), color(color_)
 {
 }
 
+void OwnPatternFormatter::formatExtendedJSON(const DB::ExtendedLogMessage & msg_ext, std::string & text) const
+{
+    DB::WriteBufferFromString wb(text);
+    const Poco::Message & msg = msg_ext.base;
+    DB::writeChar('{', wb);
+    writeCString("\"", wb);
+    writeCString("datetime", wb); // key
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    /// Change delimiters in date for compatibility with old logs.
+    DB::writeDateTimeText<'.', ':'>(msg_ext.time_seconds, wb);
+
+    DB::writeChar('.', wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 100000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 10000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 1000) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 100) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 10) % 10), wb);
+    DB::writeChar('0' + ((msg_ext.time_microseconds / 1) % 10), wb);
+    writeCString("\"", wb);
+
+    writeCString(",", wb);
+
+    writeCString("\"", wb);
+    writeCString("thread_id", wb);
+    writeCString("\"", wb);
+    writeCString(":", wb);
+    writeCString("\"", wb);
+    writeCString(" [ ", wb);
```

Comment:
There is no sense to put decorated string ` [1]` instead of number.

Reply:
Agreed. Will change this. When I found test case failing because of `assert` in python that checks specific format with `[` or `{`, I added those to make test pass.

Reply:
change submitted in updated PR.